src/main/java/org/jenkinsci/plugins/configfiles/maven/security/CredentialsHelper.java

```diff
@@ -150,17 +150,32 @@ public static String fillAuthentication(String settingsContent, Map<String, Stan
     /**
      * Removes all childs
      * 
-     * @param serverNode
+     * @param serversNode
      *            the node to remove all childs from
      */
-    private static void removeAllChilds(final Node serverNode) {
-        final NodeList childNodes = serverNode.getChildNodes();
-        for (int i = 0; i < childNodes.getLength(); i++) {
-            final Node child = childNodes.item(i);
-            serverNode.removeChild(child);
-            --i;
+    private static void removeAllChilds(final Node serversNode, final Set<String> credentialKeys, final Boolean replaceAll) {
+        final NodeList serverNodes = serversNode.getChildNodes();
+        for (int i = 0; i < serverNodes.getLength(); i++) {
+            final Node server = serverNodes.item(i);
+            String serverId = getServerId(server);
+            if (Boolean.TRUE.equals(replaceAll) || (credentialKeys.contains(serverId) && Boolean.FALSE.equals(replaceAll))) {
```

Comment:
is `&& Boolean.FALSE.equals(replaceAll)` really required? if `serverId` is containedin the key, then we remove it anyway